packages/template-block/src/components/TemplatePreview.tsx

```diff
@@ -0,0 +1,171 @@
+        textPositioning,
+    } = blockSettings;
+
+    const { previewCustom } = blockAssets;
+
+    const flexDirection = textPositioningToFlexDirection[textPositioning];
+    const isRows = flexDirection === 'row' || flexDirection === 'row-reverse';
+    const borderRadius = hasRadius_templatePreview
+        ? radiusValue_templatePreview
+        : radiusStyleMap[radiusChoice_templatePreview];
+    const border = hasBorder_templatePreview
+        ? `${borderWidth_templatePreview} ${borderStyle_templatePreview} ${toRgbaString(borderColor_templatePreview)}`
+        : 'none';
+    const height = isPreviewHeightCustom ? previewHeightCustom : previewHeightValues[previewHeightSimple];
+
+    const [isActionFlyoutOpen, setIsActionFlyoutOpen] = useState(false);
+
+    return (
+        <div
+            data-test-id="template-block-preview"
+            style={{
+                width: isRows ? `${100 - parseInt(textRatio)}%` : '100%',
+            }}
+        >
+            {template !== null || previewCustom ? (
+                <div
+                    className="tw-relative tw-overflow-hidden"
+                    data-test-id="template-block-preview-wrapper"
+                    style={{
+                        ...(hasBackgroundTemplatePreview && {
+                            ...getBackgroundColorStyles(backgroundColorTemplatePreview),
+                        }),
+                        borderRadius,
+                        border,
+                        height,
+                    }}
+                >
+                    <img
+                        data-test-id="template-block-preview-img"
+                        src={
+                            preview === PreviewType.Custom && previewCustom
+                                ? previewCustom[0].previewUrl
+                                : template?.previewUrl
+                        }
+                        className={merge([
+                            'tw-relative tw-w-full tw-h-full',
+                            `tw-object-${previewDisplayValues[previewDisplay]}`,
+                        ])}
+                        style={{
+                            objectPosition: previewImageAnchoring
+                                ? previewImageAnchoringValues[previewImageAnchoring]
+                                : 'center',
+                        }}
+                        width={preview === PreviewType.Custom && previewCustom ? previewCustom[0].width : 'auto'}
+                        height={preview === PreviewType.Custom && previewCustom ? previewCustom[0].height : 'auto'}
+                        alt={template?.name}
+                    />
+                    {isEditing && (
+                        <div className="tw-absolute tw-top-0 tw-right-0 tw-flex tw-justify-end tw-pt-3">
+                            <Flyout
+                                isOpen={isActionFlyoutOpen}
+                                trigger={
+                                    <Button
```

Comment:
And also this button is like that nested within a div with tabIndex=0, so this Element has focusable descendants which is bad for accessibility

Reply:
I also noticed it but this div comes from a fondue component and I don't have an ability to fix it directly. 
Anyway, I've checked the basic accessibility features and this component works pretty well.

Reply:
You shouldn't use the `Button` directly within the trigger. But it should be possible to remove the automatically added div around: https://beta-fondue-components.netlify.app/?path=/story/components-flyout--with-button-trigger

Reply:
Should be something like that: 
```
trigger={(_, ref) => (
                                    <Button
                                        aria-label="Template actions"
                                        emphasis={ButtonEmphasis.Default}
                                        style={ButtonStyle.Default}
                                        icon={<IconDotsVertical />}
                                        onClick={() => setIsActionFlyoutOpen((previousValue) => !previousValue)}
                                        ref={ref as MutableRefObject<HTMLButtonElement>}
                                    />
                                )}
```

Reply:
Thanks! This solution works fine 👍